src/hiscores/hiscores.module.ts

```diff
@@ -12,47 +12,30 @@ import { Bosses, Minigames, Player, Skills } from './player.model';
  * @returns {Promise<Player>}
  */
 export async function getHiscores(username: string, type: HiscoreTypes = HiscoreTypes.normal): Promise<Player> {
-    const { data } = await request<string>({
+    const { data } = await request<HiscoreResponse>({
         method: 'GET',
-        url: `http://services.runescape.com/m=${type}/index_lite.ws`,
+        url: `http://services.runescape.com/m=${type}/index_lite.json`,
         params: { player: encodeURIComponent(username) },
     });
 
-    const playerData = parseHiscoreCsv(data);
-
-    const skills = SKILL_NAMES.reduce((accumulator: Skills, skillName: string) => {
-        const [rank, level, xp] = playerData.shift();
-
-        accumulator[skillName] = { rank, level, xp };
-
-        return accumulator;
-    }, {} as Skills);
-
-    const minigames = MINIGAME_NAMES.reduce((accumulator: Minigames, minigameName: string) => {
-        const [rank, score] = playerData.shift();
-
-        accumulator[minigameName] = { rank, score };
-
-        return accumulator;
-    }, {} as Minigames);
-
-    const bosses = BOSS_NAMES.reduce((accumulator: Bosses, bossName: string) => {
-        const [rank, kills] = playerData.shift();
-
-        accumulator[bossName] = { rank, kills };
-
-        return accumulator;
-    }, {} as Bosses);
+    const skills = {} as Skills;
+    for (const skill of data.skills) {
+        const { rank, level, xp, name } = skill;
+        const skillName = SKILL_NAMES[skill.name];
+        skills[skillName] = { rank, level, xp };
+    }
+
+    const minigames = {} as Minigames;
+    const bosses = {} as Bosses;
+    for (const activity of data.activities) {
+        const { rank, score, name } = activity;
+        const isMinigame = !!MINIGAME_NAMES[name];
+        const activityName = isMinigame ? MINIGAME_NAMES[name] : BOSS_NAMES[name];
+        if (activityName) {
+            if (isMinigame) minigames[activityName] = { rank, score };
+            else bosses[activityName] = { rank, kills: score };
+        }
+    }
```

Comment:
```suggestion
    for (const activity of data.activities) {
        const { rank, score, name } = activity;

        if (!!MINIGAME_NAMES[name]) {
            minigames[MINIGAME_NAMES[name]] = { rank, score };
        } else if (!!BOSS_NAMES[name]) {
            bosses[BOSS_NAMES[name]] = { rank, kills: score };
        }
    }
```

How do you feel about this change?